plotly/tools.py

```diff
@@ -1518,33 +1518,32 @@ def _map_face2color(face, colormap, vmin, vmax):
                                          "and vmax. The vmin value cannot be "
                                          "bigger than or equal to the value "
                                          "of vmax.")
-        # find the normalized distance t of a triangle face between vmin and
-        #vmax where the distance is normalized between 0 and 1
-        t = (face - vmin) / float((vmax - vmin))
-
-        if len(colormap) <= 1:
-            t_color = colormap[0]
-            color = FigureFactory._convert_to_RGB_255(t_color)
-            color = FigureFactory._label_rgb(color)
+
+        if len(colormap) == 1:
+            # color each triangle face with the same color in colormap
+            face_color = colormap[0]
+            face_color = FigureFactory._convert_to_RGB_255(face_color)
+            face_color = FigureFactory._label_rgb(face_color)
         else:
-            # account for colormaps with more than one color
-            incr = 1./(len(colormap) - 1)
-            low_color_index = int(t/incr)
+            # find the normalized distance t of a triangle face between vmin
+            # and vmax where the distance is normalized between 0 and 1
+            t = (face - vmin) / float((vmax - vmin))
+            low_color_index = int(t / (1./(len(colormap) - 1)))
```

Comment:
🐄 Sorry if I'm just missing something, but isn't:

`1 / ( 1 / val ) = val`

And isn't `iterable[len(iterable) - 1] = iterable[-1]`

So

`iterable[1 / ( 1 / (len(iterable) - 1)] = iterable[-1]`

I feel like `iterable[-1]` is more understandable than `iterable[1 / ( 1 / (len(iterable) - 1)]`

This means that you don't need to calculate `low_color_index = int(t / (1./(len(colormap) - 1)))` until you're inside the `else` statement, which also improves readability here.